kedro/framework/session/session.py

```diff
@@ -397,9 +397,7 @@ def run(  # noqa: PLR0913
         try:
             if isinstance(runner, ThreadRunner):
                 for ds in filtered_pipeline.datasets():
-                    if catalog._match_pattern(
-                        catalog._dataset_patterns, ds
-                    ) or catalog._match_pattern(catalog._default_pattern, ds):
+                    if catalog.config_resolver.match_pattern(ds):
```

Comment:
Can this one be taken outside of the `try` without causing any problems? The look of `try-if-for-if` is not beautiful....

Reply:
yes, the only difference is that in case of the exception in this block, it won't be caught by `on_pipeline_error` hook